modules/reindex/src/main/java/org/elasticsearch/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -819,147 +815,99 @@ public static RequestWrapper<DeleteRequest> wrap(DeleteRequest request) {
     /**
      * Apply a {@link Script} to a {@link RequestWrapper}
      */
-    public abstract static class ScriptApplier implements BiFunction<RequestWrapper<?>, ScrollableHitSource.Hit, RequestWrapper<?>> {
+    public abstract static class ScriptApplier<T extends Metadata>
+        implements
+            BiFunction<RequestWrapper<?>, ScrollableHitSource.Hit, RequestWrapper<?>> {
+        protected static final String INDEX = "index";
```

Comment:
Would you please add a comment here on what this particular INDEX represents. I believe it's the op, right?

Reply:
Done.